docs/developer-experience/artificial-intelligence.md

```diff
@@ -0,0 +1,63 @@
+- __Generate Tests__. Copilot can generate unit tests for your code. Though currently this is limited to JavaScript and TypeScript.
+
+## GitHub Copilot X
+
+This extention for Copilot is currently in technical preview. It will enable the following additional features:
+
+- __Chat__. Rather than just providing code completion, Copilot will be able to have a conversation with you about what you want to do. It has context about the code you are working on and can provide suggestions based on that context.
+- __Explain__. Copilot can explain what the code is doing in natural language.
+- __Write Code__. Given prompting by the developer it can write code that you can one-click deploy into existing or new files.
+- __Debug__. Copilot can analyze your code and propose solutions to fix bugs.
+
+It can do most of what Labs can do with "brushes" as "topics", but whereas Labs changes the code in your file, the chat functionality just shows what it would change in the window. However, there is also an "inline mode" for GitHub Copilot Chat that allows you to make changes to your code inline which does not have this same limitation.
+
+You must sign-up for the preview [here](https://github.com/features/preview/copilot-x). Once you are accepted, you will be able to install the extension from the VS Code Marketplace. You may require the Insiders build of VS Code.
+
+## ChatGPT / Bing Chat
+
+Prior to GitHub Copilot X chat availability, we used ChatGPT and Bing Chat to provide similar functionality. We found the generation of code using ChatGPT to be most valuable for these scenarios:
+
+- __Write Code__. When using a library or framework that you are not familiar with, ChatGPT can help you get started by generating code.
+- __Build SQL Indexes__. Given a query, ChatGPT can generate a SQL index that will improve the performance of the query.
+- __Write Regular Expressions__. These are notoriously difficult to write, but ChatGPT can generate them for you if you give some sample input and describe what you want to extract.
+- __Improve and Validate__. If you are unsure of the implications of writing code a particular way, you can ask ChatGPT questions about it. For instance, you might ask if there is a way to write the code that is more performant or uses less memory. Once it gives you an opinion, you can ask it to provide documentation validating that assertion.
+- __Change Perspective__. ChatGPT can impersonate a persona or even a system and answer questions from that perspective. For example, you can ask it to explain what a particular piece of code does from the perspective of a user. You might have ChatGPT imagine it is a database adminstrator and ask it to explain how to improve a particular query.
+
+When using Bing Chat, we found much better results using Creative Mode.
```

Comment:
```suggestion
When using Bing Chat, experiment with modes, sometimes changing to creative mode can give the results you need.
```

Reply:
We generally try to avoid blog/story telling wording like "our dev crew bla bla..." to keep a consistent tone of voice, also note that the playbook, while written by dev crews is meant both for our crews, customers and partners, so we should keep our internal structures out of the playbook and use more imperative language.